### DISCUSSION TITLE
Icons

### DISCUSSION THREAD

**Discussion by gierens** [eza-community/eza#196] [Development]
This is a dedicated discussion about icon changes.

eza uses code points from the Unicode private use area for its icons, for an overview see: https://gist.github.com/gierens/3c7ce960eea7feb1ac53fa110f950ef6

**Comment by cfxegbert** [comment]
The current icon for `.yml` and `.yaml` files is  nf-oct-file_symlink_file (\uf481) which is not correct.   nf-seti-yml (\ue6a8) is a logical replacement but I'm not sure how the exclamation mark is related to YAML and it is very bold compared to other icons.

For comparison the icon for JSON is  nf-seti-json (\ue60b) and XML is 󰗀 nf-md-xml  (\uf05c0).

What icon should we use for YAML?

**Reply by cfxegbert** [reply]
Reading the YAML spec I see "!" is used for nodes.

**Comment by cfxegbert** [comment]
The current icon for the go language is  nf-dev-go (\ue724).  There is a filled version  nf-seti-go (\ue627) that is easier to see.  The new branding for go uses  nf-seti-go2 (\ue65e).

Should we upgrade to the filled version or to the new branding?

**Reply by PThorpe92** [reply]
I do like the filled version of the go icon :+1:

**Comment by cfxegbert** [comment]
I have been going through all the tickets and discussions for icons on `exa` and have assembled a list.  I feel some of the added files may be to specific.

A copy of this table can be found at https://gist.github.com/cfxegbert/9c502dce80eec8cf8298970902607fd7

# Added extensions
| Ext                | Icon |            Class             |      Code       | Notes                                                                |
|--------------------|:----:|:----------------------------:|:---------------:|----------------------------------------------------------------------|
| `asc`              |  󰦝  |      nf-md-shield_lock       |     \uf099d     | GnuPG ASCII armored                                                  |
| `kbx`              |  󰯄  |       nf-md-shield_key       |     \uf0bc4     | GnuPG keybox                                                         |
| `md5`              |  󰕥  |      nf-md-shield_check      |     \uf0565     | MD5 checksum                                                         |
| `sha1`             |  󰕥  |      nf-md-shield_check      |     \uf0565     | SHA-1 checksum                                                       |
| `sha224`           |  󰕥  |      nf-md-shield_check      |     \uf0565     | SHA-224 checksum                                                     |
| `sha256`           |  󰕥  |      nf-md-shield_check      |     \uf0565     | SHA-256 checksum                                                     |
| `sha384`           |  󰕥  |      nf-md-shield_check      |     \uf0565     | SHA-384 checksum                                                     |
| `sha512`           |  󰕥  |      nf-md-shield_check      |     \uf0565     | SHA-512 checksum                                                     |
| `asp`              |     |          nf-fa-code          |     \uf121      | Active server page                                                   |
| `br`               |     |       nf-oct-file_zip        |     \uf410      | Brotli compressed file                                               |
| `com`              |     |     nf-cod-terminal_cmd      |     \uebc4      | DOS command                                                          |
| `cue`              |  󰲹  | nf-md-playlist_music_outline |     \uf0cb9     | Cue sheet. Defines the way tracks are organized on CDs and SACDs.    |
| `m3u`              |  󰲹  | nf-md-playlist_music_outline |     \uf0cb9     | M3U media playlist                                                   |
| `m3u8`             |  󰲹  | nf-md-playlist_music_outline |     \uf0cb9     | M3U UTF-8 media playlist                                             |
| `dot`              |  󱁉  |         nf-md-graph          |     \uf1049     | Graphviz                                                             |
| `gv`               |  󱁉  |         nf-md-graph          |     \uf1049     | Graphviz                                                             |
| `download`         |  󰇚  |        nf-md-download        |     \uf01da     | Temporary file during downloads                                      |
| `bundle`           |     |         nf-fa-apple          |     \uf179      | Mac OS X application bundle                                          |
| `dyld`             |     |         nf-fa-apple          |     \uf179      | Directory? Mach-O dynamic library                                    |
| `plist`            |     |         nf-fa-apple          |     \uf179      | Mac OS X property list                                               |
| `elc`              |     |       nf-custom-emacs        |     \ue632      | Emacs compiled lisp                                                  |
| `heics`            |     |      nf-fa-video_camera      |     \uf03d      | https://www.loc.gov/preservation/digital/formats/fdd/fdd000526.shtml |
| `kt`               |     |       nf-custom-kotlin       |     \ue634      | Kotlin source code                                                   |
| `kts`              |     |       nf-custom-kotlin       |     \ue634      | Kotlin script                                                        |
| `ldb`              |     |        nf-fa-database        |     \uf1c0      | Microsoft Access lock file for mdb database                          |
| `mdb`              |     |        nf-fa-database        |     \uf1c0      | Microsoft Access database or Avid media database                     |
| `lisp`             |  󰅲  |    nf-md-code_parentheses    |     \uf0172     | Lisp source code                                                     |
| `p12`              |     |          nf-cod-key          |     \ueb11      | PKCS#12 certificate (Netscape)                                       |
| `pfx`              |     |          nf-cod-key          |     \ueb11      | PKCS#12 certificate (Microsoft)                                      |
| `phar`             |     |          nf-dev-php          |     \ue73d      | PHP archive                                                          |
| `pkg`              |     |        nf-cod-package        |     \ueb29      | Package                                                              |
| `pp`               |     |       nf-custom-puppet       |     \ue631      | Puppet manifest                                                      |
| `pyd`              |     |        nf-seti-python        |     \ue606      | Python dynamic module                                                |
| `pyi`              |     |        nf-seti-python        |     \ue606      | Python interface definition                                          |
| `pyo`              |     |        nf-seti-python        |     \ue606      | Python optimized code                                                |
| `whl`              |     |        nf-seti-python        |     \ue606      | Python wheel package                                                 |
| `qcow`             |     |         nf-fae-disco         |     \ue271      | QEMU copy on write disk image                                        |
| `sig`              |  󱧃  |       nf-md-file_sign        |     \uf19c3     | GnuPG signed file                                                    |
| `signature`        |  󱧃  |       nf-md-file_sign        |     \uf19c3     | GnuPG signed file                                                    |
| `srt`              |  󰨖  |       nf-md-subtitles        |     \uf0a16     | SubRip subtitle                                                      |
| `sub`              |  󰨖  |       nf-md-subtitles        |     \uf0a16     | MicroDVD subtitle                                                    |
| `sublime-build`    |     |        nf-dev-sublime        |     \ue7aa      | Sublime Text                                                         |
| `sublime-keymap`   |     |        nf-dev-sublime        |     \ue7aa      | Sublime Text                                                         |
| `sublime-menu`     |     |        nf-dev-sublime        |     \ue7aa      | Sublime Text                                                         |
| `sublime-options`  |     |        nf-dev-sublime        |     \ue7aa      | Sublime Text                                                         |
| `sublime-package`  |     |        nf-dev-sublime        |     \ue7aa      | Sublime Text                                                         |
| `sublime-project`  |     |        nf-dev-sublime        |     \ue7aa      | Sublime Text                                                         |
| `sublime-session`  |     |        nf-dev-sublime        |     \ue7aa      | Sublime Text                                                         |
| `sublime-settings` |     |        nf-dev-sublime        |     \ue7aa      | Sublime Text                                                         |
| `sublime-snippet`  |     |        nf-dev-sublime        |     \ue7aa      | Sublime Text                                                         |
| `sublime-theme`    |     |        nf-dev-sublime        |     \ue7aa      | Sublime Text                                                         |
| `tf`               |  󱁢  |       nf-md-terraform        | nf-md-terraform | Terraform                                                            |
| `tfstate`          |  󱁢  |       nf-md-terraform        | nf-md-terraform | Terraform                                                            |
| `tfvars`           |  󱁢  |       nf-md-terraform        | nf-md-terraform | Terraform                                                            |
| `ttc`              |     |          nf-fa-font          |     \uf031      | TrueType font collection                                             |
| `xcf`              |     |      nf-fa-file_image_o      |     \uf1c5      | GIMP image                                                           |

# Changed extensions
| Ext     | Icon |        Class        |  Code   | Old icon |     Old class      | Old code | Notes                                                                |
|---------|:----:|:-------------------:|:-------:|:--------:|:------------------:|:--------:|----------------------------------------------------------------------|
| `eps`   |  󰕙  | nf-md-vector_curve  | \uf0559 |         | nf-fa-file_image_o |  \uf1c5  |                                                                      |
| `ps`    |  󰕙  | nf-md-vector_curve  | \uf0559 |         | nf-fa-file_image_o |  \uf1c5  |                                                                      |
| `svg`   |  󰕙  | nf-md-vector_curve  | \uf0559 |         | nf-fa-file_image_o |  \uf1c5  |                                                                      |
| `exe`   |     | nf-cod-terminal_cmd | \uebc4  |         |   nf-fa-windows    |  \uf17a  |                                                                      |
| `gpg`   |  󰦝  |  nf-md-shield_lock  | \f099d  |         |  nf-seti-license   |  \ue60a  |                                                                      |
| `h`     |     |     nf-custom-c     | \ue61e  |         |   nf-fa-h_square   |  \uf0fd  |                                                                      |
| `hpp`   |     |    nf-custom-cpp    | \ue61d  |         |   nf-fa-h_square   |  \uf0fd  |                                                                      |
| `hxx`   |     |    nf-custom-cpp    | \ue61d  |         |   nf-fa-h_square   |  \uf0fd  |                                                                      |
| `heic`  |     | nf-fa-file_image_o  | \uf1c5  |         | nf-fa-video_camera |  \uf03d  | https://www.loc.gov/preservation/digital/formats/fdd/fdd000526.shtml |
| `image` |     |    nf-fae-disco     | \ue271  |         | nf-fa-file_image_o |  \uf1c5  |                                                                      |
| `ini`   |     |   nf-seti-config    | \ue615  |         |   nf-fa-windows    |  \uf17a  |                                                                      |
| `mk`    |     |  nf-seti-makefile   | \ue673  |         |  nf-oct-terminal   |  \uf489  | Makefile                                                             |
| `scss`  |     |    nf-seti-sass     | \ue603  |         |    nf-dev-css3     |  \ue749  |                                                                      |
| `tc`    |     |    nf-fae-disco     | \ue271  |         |  nf-oct-file_zip   |  \uf410  | TrueCrypt volume                                                     |

# Removed extensions
| Ext             | Notes                      |
|-----------------|----------------------------|
| `bashrc`        | Handled by file name table |
| `bash_history`  | Handled by file name table |
| `bash_profile`  | Handled by file name table |
| `ds_store`      | Handled by file name table |
| `gitattributes` | Handled by file name table |
| `gitignore`     | Handled by file name table |
| `gitmodules`    | Handled by file name table |
| `npmignore`     | Handled by file name table |
| `procfile`      | Not a file extension       |
| `rakefile`      | Not a file extension       |
| `zshrc`         | Handled by file name table |

# Added files
| Ext                   | Icon |          Class          |  Code   | Notes                                |
|-----------------------|:----:|:-----------------------:|:-------:|--------------------------------------|
| `.bash_history`       |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.bash_logout`        |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.bash_profile`       |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.cshrc`              |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.kshrc`              |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.login`              |  󱆃  |       nf-md-bash        | \uf1183 | (t)csh                               |
| `.logout`             |  󱆃  |       nf-md-bash        | \uf1183 | (t)csh                               |
| `.profile`            |  󱆃  |       nf-md-bash        | \uf1183 | sh                                   |
| `.tshrc`              |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.zlogin`             |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.zlogout`            |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.zprofile`           |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.zshenv`             |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.zsh_history`        |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `.zsh_sessions`       |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `bashrc`              |  󱆃  |       nf-md-bash        | \uf1183 |                                      |
| `csh.cshrc`           |  󱆃  |       nf-md-bash        | \uf1183 | /etc version                         |
| `csh.login`           |  󱆃  |       nf-md-bash        | \uf1183 | /etc version                         |
| `csh.logout`          |  󱆃  |       nf-md-bash        | \uf1183 | /etc version                         |
| `profile`             |  󱆃  |       nf-md-bash        | \uf1183 | /etc version                         |
| `zlogin`              |  󱆃  |       nf-md-bash        | \uf1183 | /etc version                         |
| `zlogout`             |  󱆃  |       nf-md-bash        | \uf1183 | /etc version                         |
| `zprofile`            |  󱆃  |       nf-md-bash        | \uf1183 | /etc version                         |
| `zshenv`              |  󱆃  |       nf-md-bash        | \uf1183 | /etc version                         |
| `zshrc`               |  󱆃  |       nf-md-bash        | \uf1183 | /etc version                         |
| `.cargo`              |     |       nf-dev-rust       | \ue7a8  | Rust package manager                 |
| `.clang-format`       |     |     nf-seti-config      | \ue615  |                                      |
| `.config`             |     | nf-custom-folder_config | \ue5fc  | Directory                            |
| `.DS_Store`           |     |       nf-fa-apple       | \uf179  | Apple custom attributes              |
| `._DS_Store`          |     |       nf-fa-apple       | \uf179  | AppleDouble dual-fork                |
| `.github`             |     | nf-custom-folder_github | \ue5fd  | Directory                            |
| `.gitlab-ci.yml`      |     |      nf-fa-gitlab       | \uf296  |                                      |
| `.htaccess`           |     |     nf-seti-config      | \ue615  |                                      |
| `.htpasswd`           |     |     nf-seti-config      | \ue615  |                                      |
| `.ideavimrc`          |     |       nf-dev-vim        | \ue7c5  | Vim setting for JetBrains vim plugin |
| `.inputrc`            |     |     nf-seti-config      | \ue615  | Keyboard mapping                     |
| `.node_repl_history`  |     |   nf-dev-nodejs_small   | \ue718  |                                      |
| `.npm`                |     |  nf-custom-folder_npm   | \ue5fa  | Directory                            |
| `.npmignore`          |     |       nf-dev-npm        | \ue71e  |                                      |
| `.npmrc`              |     |       nf-dev-npm        | \ue71e  |                                      |
| `.python_history`     |     |     nf-seti-python      | \ue606  |                                      |
| `.rustfmt.toml`       |     |       nf-dev-rust       | \ue7a8  |                                      |
| `.rvmrc`              |     |      nf-fae-ruby_o      | \ue21e  |                                      |
| `.ssh`                |  󰢬  |    nf-md-folder_key     | \uf08ac | Directory                            |
| `.viminfo`            |     |       nf-dev-vim        | \ue7c5  |                                      |
| `.Xauthority`         |     |     nf-seti-config      | \ue615  |                                      |
| `.xinitrc`            |     |     nf-seti-config      | \ue615  |                                      |
| `.Xresources`         |     |     nf-seti-config      | \ue615  |                                      |
| `.yarnrc`             |     |      nf-seti-yarn       | \ue6a7  |                                      |
| `a.out`               |     |     nf-oct-terminal     | \uf489  |                                      |
| `authorized_keys`     |  󰣀  |        nf-md-ssh        | \uf08c0 | SSH                                  |
| `bspwmrc`             |     |     nf-seti-config      | \ue615  | Tiling window manager                |
| `build.gradle.kts`    |     |     nf-seti-gradle      | \ue660  |                                      |
| `Cargo.toml`          |     |       nf-dev-rust       | \ue7a8  |                                      |
| `CMakeLists.txt`      |     |    nf-seti-makefile     | \ue673  |                                      |
| `composer.json`       |     |       nf-dev-php        | \ue73d  | PHP dependency management            |
| `composer.lock`       |     |       nf-dev-php        | \ue73d  | PHP dependency management            |
| `config.status`       |     |     nf-seti-config      | \ue615  | GNU autoconf                         |
| `configure`           |     |      nf-fa-wrench       | \uf0ad  | GNU autoconf                         |
| `configure.ac`        |     |     nf-seti-config      | \ue615  | GNU autoconf                         |
| `configure.in`        |     |     nf-seti-config      | \ue615  | GNU autoconf                         |
| `constraints.txt`     |     |     nf-seti-python      | \ue606  |                                      |
| `cron.d`              |     | nf-custom-folder_config | \ue5fc  | Directory                            |
| `cron.daily`          |     | nf-custom-folder_config | \ue5fc  | Directory                            |
| `cron.hourly`         |     | nf-custom-folder_config | \ue5fc  | Directory                            |
| `cron.monthly`        |     | nf-custom-folder_config | \ue5fc  | Directory                            |
| `cron.weekly`         |     | nf-custom-folder_config | \ue5fc  | Directory                            |
| `crontab`             |     |     nf-seti-config      | \ue615  |                                      |
| `crypttab`            |     |     nf-seti-config      | \ue615  |                                      |
| `Desktop`             |     |      nf-fa-desktop      | \uf108  | Directory                            |
| `Downloads`           |  󰉍  |  nf-md-folder_download  | \uf024d | Directory                            |
| `environment`         |     |     nf-seti-config      | \ue615  | /etc                                 |
| `etc`                 |     | nf-custom-folder_config | \ue5fc  | Directory                            |
| `GNUmakefile`         |     |    nf-seti-makefile     | \ue673  |                                      |
| `go.work`             |     |       nf-seti-go2       | \ue65e  | Go workspace                         |
| `gradle.properties`   |     |     nf-seti-gradle      | \ue660  |                                      |
| `gradlew`             |     |     nf-seti-gradle      | \ue660  |                                      |
| `gradlew.bat`         |     |     nf-seti-gradle      | \ue660  |                                      |
| `group`               |     |       nf-fa-lock        | \uf023  | /etc                                 |
| `gshadow`             |     |       nf-fa-lock        | \uf023  | /etc                                 |
| `heroku.yml`          |     |      nf-dev-heroku      | \ue77b  |                                      |
| `hostname`            |     |     nf-seti-config      | \ue615  | /etc                                 |
| `id_dsa`              |  󰌆  |        nf-md-key        | \uf0306 | SSH                                  |
| `id_ecdsa`            |  󰌆  |        nf-md-key        | \uf0306 | SSH                                  |
| `id_ecdsa_sk`         |  󰌆  |        nf-md-key        | \uf0306 | SSH                                  |
| `id_ed25519`          |  󰌆  |        nf-md-key        | \uf0306 | SSH                                  |
| `id_ed25519_sk`       |  󰌆  |        nf-md-key        | \uf0306 | SSH                                  |
| `id_rsa`              |  󰌆  |        nf-md-key        | \uf0306 | SSH                                  |
| `inputrc`             |     |     nf-seti-config      | \ue615  | /etc                                 |
| `jsconfig.json`       |     |    nf-dev-javascript    | \ue74e  |                                      |
| `Jenkinsfile`         |     |     nf-seti-jenkins     | \ue66e  |                                      |
| `Justfile`            |     |      nf-fa-wrench       | \uf0ad  |                                      |
| `known_hosts`         |  󰣀  |        nf-md-ssh        | \uf08c0 | SSH                                  |
| `LICENSE.md`          |     |       nf-fa-book        | \uf02d  |                                      |
| `LICENSE.txt`         |     |       nf-fa-book        | \uf02d  |                                      |
| `localtime`           |     |      nf-oct-clock       | \uf43a  | /etc                                 |
| `Mail`                |  󰇰  |   nf-md-email_outline   | \uf01f0 | Directory                            |
| `makefile`            |     |    nf-seti-makefile     | \ue673  |                                      |
| `Makefile.ac`         |     |    nf-seti-makefile     | \ue673  | Automake                             |
| `Makefile.in`         |     |    nf-seti-makefile     | \ue673  | Automake                             |
| `Movies`              |  󰿎  |    nf-md-movie_open     | \uf0fce | Directory                            |
| `Music`               |  󱍙  |   nf-md-folder_music    | \uf1359 | Directory                            |
| `npm-shrinkwrap.json` |     |       nf-dev-npm        | \ue71e  |                                      |
| `npmrc`               |     |       nf-dev-npm        | \ue71e  |                                      |
| `npm_cache`           |     |  nf-custom-folder_npm   | \ue5fa  | Directory                            |
| `package-lock.json`   |     |       nf-dev-npm        | \ue71e  |                                      |
| `package.json`        |     |       nf-dev-npm        | \ue71e  |                                      |
| `pam.d`               |  󰢬  |    nf-md-folder_key     | \uf08ac | Directory                            |
| `php.ini`             |     |       nf-dev-php        | \ue73d  |                                      |
| `Pictures`            |  󰉏  |   nf-md-folder_image    | \uf024f | Directory                            |
| `pom.xml`             |     |      nf-seti-maven      | \ue674  |
| `Procfile`            |     |      nf-dev-heroku      | \ue77b  |                                      |
| `pyproject.toml`      |     |     nf-seti-python      | \ue606  |                                      |
| `Rakefile`            |     |      nf-fae-ruby_o      | \ue21e  |                                      |
| `release.toml`        |     |       nf-dev-rust       | \ue7a8  |                                      |
| `requirements.txt`    |     |     nf-seti-python      | \ue606  |                                      |
| `robots.txt`          |  󰚩  |       nf-md-robot       | \uf06a9 |                                      |
| `rvmrc`               |     |      nf-fae-ruby_o      | \ue21e  | /etc                                 |
| `settings.gradle.kts` |     |     nf-seti-gradle      | \ue660  |                                      |
| `shadow`              |     |       nf-fa-lock        | \uf023  | /etc                                 |
| `shells`              |     |     nf-seti-config      | \ue615  | /etc                                 |
| `ssh`                 |  󰢬  |    nf-md-folder_key     | \uf08ac | Directory                            |
| `sudoers`             |     |       nf-fa-lock        | \uf023  | /etc                                 |
| `sudoers.d`           |  󰢬  |    nf-md-folder_key     | \uf08ac | Directory                            |
| `timezone`            |     |      nf-oct-clock       | \uf43a  | /etc                                 |
| `tsconfig.json`       |     |   nf-seti-typescript    | \ue628  |                                      |
| `Videos`              |     |   nf-fa-video_camera    | \uf03d  | Directory                            |
| `webpack.config.js`   |  󰜫  |      nf-md-webpack      | \uf072b |                                      |
| `xbps.d`              |     | nf-custom-folder_config | \ue5fc  | XBPS package system                  |
| `xorg.conf.d`         |     | nf-custom-folder_config | \ue5fc  |                                      |


# Changed files
| Ext            | Icon |        Class         |  Code   | Old icon |      Old class      | Old code | Notes                          |
|----------------|:----:|:--------------------:|:-------:|:--------:|:-------------------:|:--------:|--------------------------------|
| `.bashrc`      |  󱆃  |      nf-md-bash      | \uf1183 |         |   nf-oct-terminal   |  \uf489  |                                |
| `.zshrc`       |  󱆃  |      nf-md-bash      | \uf1183 |         |   nf-oct-terminal   |  \uf489  |                                |
| `.git`         |     | nf-custom-folder_git | \ue5fb  |         |      nf-fa-git      |  \uf1d3  | Change to a git directory icon |
| `hidden`       |  󱞞  | nf-md-folder_hidden  | \uf179e |         |     nf-fa-lock      |  \uf023  | Directory                      |
| `Makefile`     |     |   nf-seti-makefile   | \ue673  |         |   nf-oct-terminal   |  \uf489  |                                |
| `node_modules` |     | nf-custom-folder_npm | \ue5fa  |         | nf-dev-nodejs_small |  \ue718  | Directory                      |
| `passwd`       |     |      nf-fa-lock      | \uf023  |         |   nf-seti-config    |  \ue615  |                                |

# Removed files
| Ext                | Notes             |
|--------------------|-------------------|
| `.bashprofile`     | .bash_profile     |
| `config`           | .config           |
| `ds_store`         | .DS_Store         |
| `gitignore_global` | .gitignore_global |
| `gitlab-ci.yml`    | .gitlab-ci.yml    |
| `npmignore`        | .npmignore        |

**Comment by corrydiehl** [comment]
hey folks - new-ish to eza. Is there a way to troubleshoot why an icon shows up as a question mark?

I'm on Mac OSX - Ventura
iTerm2
JetBrains NerdFont

Icons in particular that don't display are Go/Golang icons - likely more - this just stands out to me in my Go repos

**Reply by PThorpe92** [reply]
Hmm. Most of the time it's a problem with someone having a nerd font installed that is still at Version 2. 

So all the other ones are fine that you know of, except the .go icons? That is kinda strange

**Reply by corrydiehl** [reply]
To be fair no - the .go icons were the first I noticed but I ran across this page and there are many that show as squares with ? inside

https://gist.github.com/gierens/3c7ce960eea7feb1ac53fa110f950ef6

ETA: The list is larger, but this is what it looks like inside neovim
![image](https://github.com/eza-community/eza/assets/4060907/b0a99d6b-c6fb-461b-8be0-32d57685497a)

**Reply by PThorpe92** [reply]
EDIT: Yeah I would definitely try to re-install the latest NerdFonts icons package and see if that helps. Especially if you are missing a bunch of those. I think it's very likely you have an out of date NerdFont or are missing one of their packages.

**Reply by corrydiehl** [reply]
Thanks much, will give that a try

**Reply by corrydiehl** [reply]
That didn't cover them all - but did help ALOT - thanks for the suggestion

**Comment by rrpolanco** [comment]
I have the latest NerdFonts `3.1.1` installed and `eza` does not show the correct symbols for rust and go files:
<img width="325" alt="image" src="https://github.com/eza-community/eza/assets/6497491/c5ed0314-c21d-49b3-a5d4-55e571eedc97">

Any help to debug is appreciated.

**Reply by Omnikron13** [reply]
Sure your terminal is configured properly to use the font? I see the R in a cog rust icon for `.rs` files, which seems to be mapped to the codepoint `U+E68B`. What codepoint is the glyph you're getting?

**Comment by doughsay** [comment]
Why are JSON files the Debian logo?
![image](https://github.com/user-attachments/assets/1c2a3926-5d76-4534-9b02-7e9928273602)

**Reply by doughsay** [reply]
I'm not sure why some of the icons were working and some weren't, but explicitly installing a nerd font and explicitly configuring that font in my terminal fixed it.  I never installed any nerd font before, so I'm wondering why it was even working to begin with.

actual steps used on arch linux:

1. installed `ttf-firacode-nerd`
2. added to `alacritty.toml`:
```
[font.normal]
family = "FiraCode Nerd Font Mono"
```

**Reply by gierens** [reply]
The reason this was "sort of" working before is this: all these icons eza and other projects use are simply codepoints in the unicode private use area that nerd fonts use in a certain way. However this is no official standard, so other fonts may do other things with these codepoints.
